File: example/fs/Samples/BesselBeam.fs
namespace Samples

open System
open AUTD3Sharp
open AUTD3Sharp.Gain
open AUTD3Sharp.Modulation
open AUTD3Sharp.Utils
open type AUTD3Sharp.Units

module BesselBeamTest =
    let Test<'T> (autd : Controller<'T>) = 
        (Silencer.Default()) |> autd.Send;

        let m = new Sine (150u * Hz);

        let start = autd.Geometry.Center;
        let dir = Vector3d.UnitZ;

        let g = new Bessel(start, dir, 13.0 / 180.0 * Math.PI);
        (m, g) |> autd.Send;

File: example/Samples/BesselBeam.cs
using AUTD3Sharp;
using AUTD3Sharp.Gain;
using AUTD3Sharp.Modulation;
using AUTD3Sharp.Utils;
using static AUTD3Sharp.Units;

namespace Samples;

internal static class BesselBeamTest
{
    public static void Test(Controller autd)
    {
        var config = new Silencer();
        autd.Send(config);

        var m = new Sine(freq: 150 * Hz, option: new SineOption());

        var start = autd.Center();
        var dir = Vector3.UnitZ;
        var g = new Bessel(start, dir, 13.0f / 180.0f * MathF.PI * rad, new BesselOption());

        autd.Send((m, g));
    }
}
